lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -684,6 +684,330 @@ def test_toc_toy_from_section(self, default_ms, setup_finds, setup_sends, toc_fi
+        ),
+        (
+            'verified',
+            False,
+            None,
+            {
+                'status': 'eligible',
+                'short_description': 'Proctored Option Available',
+                'suggested_icon': 'fa-lock',
+                'in_completed_state': False
+            }
+        ),
+        (
+            'verified',
+            False,
+            'declined',
+            {
+                'status': 'declined',
+                'short_description': 'Taking As Open Exam',
+                'suggested_icon': 'fa-unlock',
+                'in_completed_state': False
+            }
+        ),
+        (
+            'verified',
+            False,
+            'submitted',
+            {
+                'status': 'submitted',
+                'short_description': 'Pending Session Review',
+                'suggested_icon': 'fa-spinner fa-spin',
+                'in_completed_state': True
+            }
+        ),
+        (
+            'verified',
+            False,
+            'verified',
+            {
+                'status': 'verified',
+                'short_description': 'Passed Proctoring',
+                'suggested_icon': 'fa-check',
+                'in_completed_state': True
+            }
+        ),
+        (
+            'verified',
+            False,
+            'rejected',
+            {
+                'status': 'rejected',
+                'short_description': 'Failed Proctoring',
+                'suggested_icon': 'fa-exclamation-triangle',
+                'in_completed_state': True
+            }
+        ),
+        (
+            'verified',
+            False,
+            'error',
+            {
+                'status': 'error',
+                'short_description': 'Failed Proctoring',
+                'suggested_icon': 'fa-exclamation-triangle',
+                'in_completed_state': True
+            }
+        ),
+    )
+    @ddt.unpack
+    def test_proctored_exam_toc(self, enrollment_mode, is_practice_exam,
+                                attempt_status, expected):
+        """
+        Generate TOC for a course with a single chapter/sequence which contains proctored exam
+        """
+        with self.store.default_store(ModuleStoreEnum.Type.mongo):
+            usage_key = self.course_key.make_usage_key('videosequence', 'Toy_Videos')
+            sequence = self.modulestore.get_item(usage_key)
+
+            sequence.is_time_limited = True
+            sequence.is_proctored_enabled = True
+            sequence.is_practice_exam = is_practice_exam
+
+            self.modulestore.update_item(sequence, self.user.id)
+
+            self.toy_course = self.modulestore.get_course(self.course_key)
+
+            # refresh cache after update
+            self.field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+                self.course_key, self.request.user, self.toy_course, depth=2
+            )
+
+            from edx_proctoring.api import (
```

Comment:
Is it not possible to put these imports at the top of the file? If at all possible, we want to avoid circular references.